internal/utils/aggregator.go

```diff
@@ -147,3 +155,51 @@ func determine_k8s_object_type(df dataframe.DataFrame) dataframe.DataFrame {
 	df = df.Mutate(s.Col("X1")).Rename("k8s_object_name", "X1")
 	return df
 }
+
+func check_if_all_required_columns_in_CSV(df dataframe.DataFrame) error {
+	// Check if all the required columns are present in CSV
+	all_required_columns := []string{
+		"report_period_start",
+		"report_period_end",
+		"interval_start",
+		"interval_end",
+		"container_name",
+		"pod",
+		"owner_name",
+		"owner_kind",
+		"workload",
+		"workload_type",
+		"namespace",
+		"image_name",
+		"node",
+		"resource_id",
+		"cpu_request_container_avg",
+		"cpu_request_container_sum",
+		"cpu_limit_container_avg",
+		"cpu_limit_container_sum",
+		"cpu_usage_container_avg",
+		"cpu_usage_container_min",
+		"cpu_usage_container_max",
+		"cpu_usage_container_sum",
+		"cpu_throttle_container_avg",
+		"cpu_throttle_container_max",
+		"cpu_throttle_container_sum",
+		"memory_request_container_avg",
+		"memory_request_container_sum",
+		"memory_limit_container_avg",
+		"memory_limit_container_sum",
+		"memory_usage_container_avg",
+		"memory_usage_container_min",
+		"memory_usage_container_max",
+		"memory_usage_container_sum",
+		"memory_rss_usage_container_avg",
+		"memory_rss_usage_container_min",
+		"memory_rss_usage_container_max",
+		"memory_rss_usage_container_sum",
+	}
+	cloumns_in_csv := df.Names()
+	if !reflect.DeepEqual(all_required_columns, cloumns_in_csv) {
```

Comment:
Before doing the DeepEqual do you think we can just compare if the cloumns_in_csv is exactly having 37 number of items? if its not there then just error out? I don't know if this is already handled in DeepEqual, however I feel this may save sometime for us, considering we are doing this for every CSV?

Apart from that do you think we would ever need to know what are missing columns if in case of error?

Reply:
Okay, looking further into it, it seems DeepEqual checks the length: https://github.com/goccy/go-reflect/blob/4646ad15ec8afb955a3bfb6af969f007f169f383/reflect.go#L364

Reply:
Looks like it is already handled in DeepEqual - https://github.com/golang/go/blob/master/src/reflect/deepequal.go#L102-#L103